### PR TITLE
Changes debug vars to be adapters over uint32_t and adds basic tests

### PR DESCRIPTION
**Pull request by jkff**
They are always 32-bit, at least for now. This way they don't need to do error checking - that belongs in the higher-level comms protocol code; and the code generally becomes more concise.

Also renames a few functions/variables to follow our more common style and abbreviates the repetitive constructors.

### CODE REVIEW

**Comment by jkff** [comment]
Oops forgot to add tests - in this PR I can add tests both to DebugVar itself and to VarCmd (their interaction seems very error-prone). Please do not approve until I do so, but feel free to comment on the rest.

File: controller/lib/debug/vars.h [hunk 1]
```diff
@@ -46,71 +46,59 @@ class DebugVar {
...
   DebugVar(const char *name, int32_t *data, const char *help = "",
-           const char *fmt = "%d");
+           const char *fmt = "%d")
+      : DebugVar(VarType::INT32, name, data, help, fmt) {}
 
   // Like above, but unsigned
   DebugVar(const char *name, uint32_t *data, const char *help = "",
-           const char *fmt = "%d");
+           const char *fmt = "%u")
+      : DebugVar(VarType::INT32, name, data, help, fmt) {}
```
Comment (jlebar) [inline controller/lib/debug/vars.h:55]: VarType::UINT32

**Comment by jlebar** [comment]
> Please do not approve until I do so

OK, un-approving :)

**Comment by jkff** [comment]
Okay, I added tests, please take another look.

File: controller/test/debug/var_cmd_test.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  uint32_t new_value = 0xCAFEBABE;
+  std::array<uint8_t, 4> new_bytes;
+  u32_to_u8(new_value, new_bytes.data());
+
+  // Test that a SET command changes the variable's value.
+  VarCmd cmd;
+  uint8_t id[2];
+  u16_to_u8(var.GetId(), id);
+  std::array set_cmd = {
```
Comment (jlebar) [inline controller/test/debug/var_cmd_test.cpp:57]: ...wait, this works if you leave off the template?

Comment (jkff) [reply controller/test/debug/var_cmd_test.cpp:57]: Yup, as of C++17 https://en.cppreference.com/w/cpp/language/class_template_argument_deduction

**Pull request merged**